packages/components-angular/tests/e2e/specs/provider.e2e.ts

```diff
@@ -0,0 +1,18 @@
+import type { ElementHandle, Page } from 'puppeteer';
+import { goto } from '../helpers';
+import type { Theme } from '@porsche-design-system/components-react';
+
+let page: Page;
+beforeEach(async () => (page = await browser.newPage()));
+afterEach(async () => await page.close());
+
+it('should change p-button theme to dark', async () => {
+  await goto(page, 'button');
+
+  const button = await page.$('p-button');
+  const getTheme = (handle: ElementHandle<Element>): Promise<Theme> => handle.evaluate((el) => (el as any).theme);
+  expect(await getTheme(button)).toBe('light');
+
+  await page.select('select[name="theme"]', 'dark');
```

Comment:
Since there are only three options, maybe test all of them here? Could also test the functionality in isolation since theres not guarantee the select is using the provider to set the theme

Reply:
I think testing every option doesn't bring additional value.  
How would you test the functionality in isolation?

Reply:
Basically constructing the page within the test